src/o365/spo/commands/list/list-view-list.spec.ts

```diff
@@ -0,0 +1,2260 @@
+import appInsights from '../../../../appInsights';
+import auth, { Site } from '../../SpoAuth';
+const command: Command = require('./list-view-list');
+import * as assert from 'assert';
+import * as request from 'request-promise-native';
+import Utils from '../../../../Utils';
+
+describe(commands.LIST_VIEW_LIST, () => {
+  let vorpal: Vorpal;
+  let log: any[];
+  let cmdInstance: any;
+  let cmdInstanceLogSpy: sinon.SinonSpy;
+  let trackEvent: any;
+  let telemetry: any;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(auth, 'getAccessToken').callsFake(() => { return Promise.resolve('ABC'); });
+    trackEvent = sinon.stub(appInsights, 'trackEvent').callsFake((t) => {
+      telemetry = t;
+    });
+  });
+
+  beforeEach(() => {
+    vorpal = require('../../../../vorpal-init');
+    log = [];
+    cmdInstance = {
+      log: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    cmdInstanceLogSpy = sinon.spy(cmdInstance, 'log');
+    auth.site = new Site();
+    telemetry = null;
+  });
+
+  afterEach(() => {
+    Utils.restore([
+      vorpal.find,
+      request.get,
+      request.post
+    ]);
+  });
+
+  after(() => {
+    Utils.restore([
+      appInsights.trackEvent,
+      auth.getAccessToken,
+      auth.restoreAuth,
+      request.get,
+      request.post
+    ]);
+  });
+
+  it('has correct name', () => {
+    assert.equal(command.name.startsWith(commands.LIST_VIEW_LIST), true);
+  });
+
+  it('has a description', () => {
+    assert.notEqual(command.description, null);
+  });
+
+  it('calls telemetry', (done) => {
+    cmdInstance.action = command.action();
+    cmdInstance.action({ options: {} }, () => {
+      try {
+        assert(trackEvent.called);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('logs correct telemetry event', (done) => {
+    cmdInstance.action = command.action();
+    cmdInstance.action({ options: {} }, () => {
+      try {
+        assert.equal(telemetry.name, commands.LIST_VIEW_LIST);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('aborts when not connected to a SharePoint site', (done) => {
+    auth.site = new Site();
+    auth.site.connected = false;
+    cmdInstance.action = command.action();
+    cmdInstance.action({ options: { debug: false, webUrl: 'https://contoso.sharepoint.com/sites/ninja', title: 'Documents' } }, (err?: any) => {
+      try {
+        assert.equal(JSON.stringify(err), JSON.stringify(new CommandError('Log in to a SharePoint Online site first')));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('retrieves all views of the specific list if listTitle option is passed (debug)', (done) => {
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url.indexOf(`https://contoso.sharepoint.com/sites/ninja/_api/web/lists/GetByTitle('Documents')/views`) > -1) {
```

Comment:
If you'd like to be really specific with full url. This line should actually be:
`if (opts.url.indexOf('https://contoso.sharepoint.com/sites/ninja/_api/web/lists/getByTitle(\'Documents\')/views') > -1) {`
instead
`if (opts.url.indexOf('https://contoso.sharepoint.com/sites/ninja/_api/web/lists/getByTitle('Documents')/views') > -1) {`

Difference is `\'Documents\'` <==> `'Documents'`
I was able to discover this by placing console log above line `462` like: 
`cmdInstanceLogSpy.getCalls().forEach(x => {console.log(x.args);});`

Console log is sometimes useful to understand what is happening. The output can be found in the tests execution like:

```
spo list view list
    √ has correct name
    √ has a description
    √ calls telemetry
    √ logs correct telemetry event
    √ aborts when not connected to a SharePoint site
[ 'Retrieving access token for https://contoso.sharepoint.com...' ]
[ 'Retrieved access token ABC. Retrieving request digest...' ]
[ 'Response:' ]
[ { FormDigestValue: 'ABC' } ]
[ '' ]
[ 'Executing web request...' ]
[ { url: 'https://contoso.sharepoint.com/sites/ninja/_api/web/lists/getByTitle(\'Documents\')/views',
    method: 'GET',
    headers:
     { authorization: 'Bearer ABC',
       accept: 'application/json;odata=nometadata',
       'User-Agent': 'NONISV|SharePointPnP|Office365CLI/1.12.0' },
    json: true } ]
[ '' ]
[ '\u001b[32mDONE\u001b[39m' ]
    1) retrieves all views of the specific list if listTitle option is passed (debug)
    2) retrieves all views of the specific list if listTitle option is passed
    3) retrieves all views of the specific list if listId option is passed (debug)
    4) retrieves all views of the specific list if listId option is passed
    5) outputs all properties when output is JSON
    √ command correctly handles list get reject request
    √ uses correct API url when listTitle option is passed
    √ uses correct API url when listId option is passed
    √ fails validation if the webUrl option not specified
    √ fails validation if both listId and listTitle options are not passed
    √ fails validation if the webUrl option is not a valid SharePoint site URL
    √ passes validation if the webUrl option is a valid SharePoint site URL
    √ fails validation if the listId option is not a valid GUID
    √ passes validation if the listId option is a valid GUID
    √ fails validation if both listId and listTitle options are passed
    √ supports debug mode
    √ has help referring to the right command
    √ has help with examples
    √ correctly handles lack of valid access token
```